lib/molinillo/dependency_graph.rb

```diff
@@ -248,7 +259,12 @@ def hash
       # dependency graph?
       # @return true iff there is a path following edges within this {#graph}
       def path_to?(other)
-        successors.include?(other) || successors.any? { |v| v.path_to?(other) }
+        walk = proc do |vertex|
+          return true if vertex.equal?(other)
+          vertex.successors.each(&walk)
+          false
+        end
+        walk.call(self)
```

Comment:
the same reason it's a `proc` instead of a `lambda` -- incredibly easy to just `return true` and stop walking